scipy/stats/_stats_py.py

```diff
@@ -10165,6 +10165,136 @@ def first_order(t):
+    j = np.arange(n, dtype=x.dtype)
+    n = np.asarray(n, dtype=x.dtype)[()]
+    return (np.sum(special.binom(j, r[:, np.newaxis])*x, axis=-1)
+            / special.binom(n-1, r) / n)
+
+
+def _prk(r, k):
+    return (-1)**(r-k)*special.binom(r, k)*special.binom(r+k, k)
+
+
+@_axis_nan_policy_factory(  # noqa: E302
+    _moment_result_object, n_samples=1, result_to_tuple=lambda x: (x,),
+    n_outputs=lambda kwds: _moment_outputs(kwds, [1, 2, 3, 4])
+)
+def lmoment(sample, order=None, *, axis=0, sorted=False, standardize=True):
+    r"""Compute L-moments of a sample from a continuous distribution
+
+    The L-moments of a probability distribution are summary statistics with
+    uses similar to those of conventional moments, but they are defined in
+    terms of the expected values of order statistics.
+    Sample L-moments are defined analogously to population L-moments, and
+    they can serve as estimators of population L-moments. They tend to be less
+    sensitive to extreme observations than conventional moments.
+
+    Parameters
+    ----------
+    sample : array_like
+        The real-valued sample whose L-moments are desired.
+    order : array_like, optional
+        The (positive integer) orders of the desired L-moments.
+        Must be a scalar or non-empty 1D array. Default is [1, 2, 3, 4].
+    axis : int, default=0
+        The axis along which to compute L-moments.
+    sorted : bool, default=False
+        Whether `sample` is already sorted in increasing order along `axis`.
+        If False (default), `sample` will be sorted.
+    standardize : bool, default=True
+        Whether to return L-moment ratios for orders 3 and higher.
+        L-moment ratios are analogous to standardized conventional
+        moments: they are the non-standardized L-moments divided
+        by the L-moment of order 2.
+
+    Returns
+    -------
+    lmoments : ndarray
+        The sample L-moments of order `order`.
+
+    See Also
+    --------
+    moment
+
+    References
+    ----------
+    .. [1] D. Bilkova. "L-Moments and TL-Moments as an Alternative Tool of
+           Statistical Data Analysis". Journal of Applied Mathematics and
+           Physics. 2014. :doi:`10.4236/jamp.2014.210104`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> rng = np.random.default_rng(328458568356392)
+    >>> sample = rng.exponential(size=100000)
+    >>> stats.lmoment(sample)
+    array([1.00124272, 0.50111437, 0.3340092 , 0.16755338])
+
+    Note that the first four standardized population L-moments of the standard
+    exponential distribution are 1, 1/2, 1/3, and 1/6; the sample L-moments
+    provide reasonable estimates.
+
+    """
+    args = lmoment_iv(sample, order, axis, sorted, standardize)
+    sample, order, axis, sorted, standardize = args
+
+    n_moments = np.max(order)
+    k = np.arange(n_moments, dtype=sample.dtype)
+    prk = _prk(np.expand_dims(k, list(range(1, sample.ndim+1))), k)
```

Comment:
OK, I can do that. I'm also a NumPy maintainer, though, so I will forgive myself : )

In the input validation of `expand_dims`, they are both allowed:

https://github.com/numpy/numpy/blob/2f7fe64b8b6d7591dd208942f1cc74473d5db4cb/numpy/lib/_shape_base_impl.py#L601-L602

And either way, the `axis` argument goes through a list comprehension before being converted back to a tuple:

https://github.com/numpy/numpy/blob/2f7fe64b8b6d7591dd208942f1cc74473d5db4cb/numpy/_core/numeric.py#L1432-L1439

Anyway, sometimes for shape manipulations can benefit from the shape being represented as a `list`, which is where the habit of using lists can creep in. That's not needed here, so I'll make the change.

Reply:
Ah I didn't know that about list input; good to know :)

The reason I mentioned this is, is because I'm currently working on adding support for shape-typing in numpy, and in practice that'll work best if `tuple` types are used for shapes. 
This is because type-checkers are able to infer the precise length (i.e. `ndim`) and contents of a `tuple`. So type-checkers "see" `(42,)` as `tuple[Literal[42]]`.
But for lists, this isn't the case. For instance `[42]` will be interpreted as `list[int]`.
So this means that in order to have proper type-hinting for the shapes of arrays, then you're gonna have to use tuples when specifying shapes. I mean "proper" as in `ndarray[tuple[42], SomeDType]`, as opposed to `ndarray[tuple[int, ...], SomeDType]`.
You can see the difference in action in this [pyright playground example](https://basedpyright.com/?typeCheckingMode=strict&code=E4UwbiCGA2D6AuBPADiAFGgLAJgDQEp8AoUCGBFdAbRwF18g).

Reply:
> If it's lists are better for performance, then by all means, go for it.

I don't have a performance argument for using them - if there is one, it would be completely negligible. Style is more important here.

Reply:
Adjusted as requested (locally; will push soon), so I'll go ahead and resolve.